toolchain/check/import_ref.cpp

```diff
@@ -712,14 +735,17 @@ class ImportRefResolver {
           }
           case SemIR::BindSymbolicName::Kind: {
             // We already imported a constant value for this symbolic binding.
-            // We can reuse most of it.
-            auto new_bind_inst_id = GetLocalConstantInstId(bind_id);
+            // We can reuse most of it, but update the value to point to our
+            // specific parameter, and preserve the constant value.
             auto new_bind_inst =
                 context_.insts().GetAs<SemIR::BindSymbolicName>(
-                    new_bind_inst_id);
+                    context_.constant_values().GetInstId(
+                        param_data.bind_const_id));
```

Comment:
Can bind_const_id ever point at an error? i.e., somewhat concerned that this might fail on an import of an interface with an invalid parameter. How tested are invalid generic imports?

Reply:
If we get here then in the imported IR we had a `BindSymbolicName`, and the `bind_const_id` is the result of importing that `BindSymbolicName`, which can't produce an error. We only support a few very specific instruction patterns here, and hit a fatal error a few lines below if we see anything not matching those patterns (unchanged in this PR).

To your broader point, I don't think we have good checking for imports of invalid generics. Is that something we're generally testing and expecting to work currently? I can see tooling use cases where it'd be important, but it's not something I've really been thinking about.

Reply:
The important thing to me is really just that check shouldn't crash. Imports of invalid generics shouldn't cause a crash. I've added tests for some various invalid import cases. I don't think you've added tests for that as part of adding generics.